src/RadhydroSimulation.hpp

```diff
@@ -570,14 +578,102 @@ template <typename problem_t> void RadhydroSimulation<problem_t>::applyPoissonGr
 	});
 }
 
+// to ensure that physical quantities are within reasonable
+// floors and ceilings which can be set in the param file
+template <typename problem_t>
+void RadhydroSimulation<problem_t>::EnforceLimits(amrex::Real const densityFloor, amrex::Real const pressureFloor, amrex::Real const speedCeiling,
```

Comment:
Can you put this in `HydroSystem`? `EnforceDensityFloor` was a function within `HydroSystem` and I think that's the right place for this as well.